integration-tests/main_test.go

```diff
@@ -29,5 +31,10 @@ func TestMain(m *testing.M) {
 		os.Exit(1)
 	}
 
+	go func() {
```

Comment:
We don't need this all the time. Also tests should be able to define the handler for themselves.

Reply:
I've added some int-test changes by default... This is a different task, so I'll remove all excess logic.